### PR TITLE
Remove some cells from Config

### PR DESCRIPTION
**Pull request by derekdreery**
In most cases it makes sense to not use interior mutability in Config, so that mutability is more transparent. This PR removes Cells in these cases. It leaves those cases where parameters are only initialized on first access (LazyCell).

### CODE REVIEW

**Comment by derekdreery** [comment]
The doc comments on `frozen` and `locked` could be improved, could anyone suggest what it should say?

**Comment by matklad** [comment]
LGTM, but let's ask @alexcrichton as well!

File: src/cargo/util/config.rs [hunk 1]
```diff
@@ -49,11 +49,16 @@ pub struct Config {
     /// The location of the rustdoc executable
     rustdoc: LazyCell<PathBuf>,
     /// Whether we are printing extra verbose messages
-    extra_verbose: Cell<bool>,
-    frozen: Cell<bool>,
-    locked: Cell<bool>,
+    extra_verbose: bool,
+    /// TODO to do with Cargo.lock
```
Comment (alexcrichton) [inline src/cargo/util/config.rs]: This has to do with the `--frozen` flag being passed on the command line, namely we're not allowed to touch the network

File: src/cargo/util/config.rs [hunk 2]
```diff
@@ -49,11 +49,16 @@ pub struct Config {
...
+    frozen: bool,
+    /// TODO to do with Cargo.lock
```
Comment (alexcrichton) [inline src/cargo/util/config.rs]: This has to do with the `--locked` flag, namely we're not allowed to change lock files

**Comment by derekdreery** [comment]
Thanks I'll update

**Pull request merged**